### PR TITLE
🔥🚀 feat: CDN (Firebase) & feat: account section

### LINKED ISSUE #910: Enhancement: Cache images (and/or serve with CDN)
### Contact Details

_No response_

### What happened?

Temporary images created by plugins can expire. This could be fixed with an option to cache returned images locally or some other option.

### Steps to Reproduce

Prompt a plugin to respond with an image.

### What browsers are you seeing the problem on?

_No response_

### Relevant log output

_No response_

### Screenshots

image

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #1196: Enhancement: Profile picture for users
### Contact Details

_No response_

### What features would you like to see added?

It is awsome to provide a a place to use their own profile picture, even just using imgur or any other picture bed websites

### More details

Should be available to set in settings

### Which components are impacted by your request?

_No response_

### Pictures

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by berry-13**
## Summary

I have implemented CDN support using Firebase. Currently, it's specifically designed for uploading profile pictures, but we plan to extend its functionality to support images within the chat and file uploads as well. Additionally, there have been some minor graphical changes, including new icons, and a reorganization of folders in the Settings Tabs. The CDN for profile pictures is currently working for Google, Facebook, Github, and Discord (OpenID support is not yet implemented). Before being uploaded, the images undergo transformation and conversion into the webp format. On average, a profile picture weighs around 3kb, with some as low as 300B. The upload profile picture functionality works for those who don't use Firebase as well. Also, DALL-E plugin support firebase

Here are some frontend images showcasing the image upload

image

image

https://github.com/danny-avila/LibreChat/assets/81851188/d1ed5973-1546-4101-a112-bbb1715aef57

### fixed some bugs in the settings tab when the theme was light

image

## Change Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Style update
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update 

## Testing
...

- tested automatic disable "auto avatar upload with social logins":  Google, Facebook, Github and Discord
- refactor login tested with Google, Facebook, Github and Discord

## Checklist

- [x] My code adheres to this project's style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented in any complex areas of my code
- [x] I have made pertinent documentation changes
- [x] My changes do not introduce new warnings
- [x] I have written tests demonstrating that my changes are effective or that my feature works
- [x] Local unit tests pass with my changes
- [x] Any changes dependent on mine have been merged and published in downstream modules.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -82,4 +82,6 @@ data.ms/*
 auth.json
 
 /packages/ux-shared/
-/images
\ No newline at end of file
+/images
+
+!client/src/components/Nav/SettingsTabs/Data/
```
Comment (berry-13) [inline .gitignore:87]: I have reorganized all the settings tabs, but it seems that "data/" is being ignored. Therefore, I added `!client/src/components/Nav/SettingsTabs/Data/` to include `Data.tsx` in commits

**Comment by danny-avila** [comment]
Did you create new backend tests? I believe you need to import the mocked logger, there are some tests that do this already not sure which.

> TypeError: path.basename is not a function

**Comment by danny-avila** [comment]
> Did you create new backend tests? I believe you need to import the mocked logger, there are some tests that do this already not sure which.
> 
> > TypeError: path.basename is not a function

@Berry-13 I would like to merge this soon so I will make edits from here if you are mostly done.

**Comment by Berry-13** [comment]
> > Did you create new backend tests? I believe you need to import the mocked logger, there are some tests that do this already not sure which.
> > 
> > > TypeError: path.basename is not a function
> 
> @Berry-13 I would like to merge this soon so I will make edits from here if you are mostly done.

sure, thank you!

**Comment by danny-avila** [comment]
My /api review is done, I just want to fix a few things on the frontend now

**Comment by danny-avila** [comment]
Noteable changes in the backend outside of the main firebase update:

- DALL-E/DALL-E-3 now uses the image basename from a url to test whether the DALL-E did correctly generate an image.

This fixes an issue where a reverse proxy may generate a different image url like so: media.mandrillai.tech/herearerandomcharscters.jpg

- When tools are loaded, they will pass the userId in case they are needed, such as the case with DALL-E

@Berry-13 following are some todos, which are not needed to merge the PR but will be needed some time after

- TODO: do not use Model directly, such as when you use `User.findById`, instead use a service method that uses the model; for example, we need a method similar to `getMessages` but instead `getUserById`
     - this is done incorrectly a few other places in the codebase (not by you)
- TODO: JSDocs are needed for all new functions. Should be typed out as much as possible.
- TODO: stable diffusion needs the same treatment as the DALL-E plugin

File: client/src/components/Nav/SettingsTabs/Account/Avatar.tsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+      if (!input) {
+        throw new Error('Nessun file selezionato');
+      }
+
+      const formData = new FormData();
+      formData.append('userId', userId);
+      formData.append('input', input, input.name);
+      formData.append('manual', 'true');
+
+      const response = await fetch('/api/files/images/avatar', {
```
Comment (danny-avila) [inline client/src/components/Nav/SettingsTabs/Account/Avatar.tsx]: we do not want to make raw fetch calls, I'll be replacing this with react-query

**Comment by danny-avila** [comment]
Noteable changes in the frontend outside of the main firebase/profile picture update:

- User object now uses recoil atom state instead of useState.

Allows changing the avatar URL to dynamically render across the app once uploaded. I've observed no issues with this change as AuthContext still handles the user object the same.

- Toast has a new z-index order of 1000 to show on top of dialog overlays

**Review by danny-avila — APPROVED** [review]

**Pull request merged**